talismane_core/src/main/java/com/joliciel/talismane/AnnotationObserver.java

```diff
@@ -0,0 +1,39 @@
+///////////////////////////////////////////////////////////////////////////////
+//Copyright (C) 2016 Joliciel Informatique
+//
+//This file is part of Talismane.
+//
+//Talismane is free software: you can redistribute it and/or modify
+//it under the terms of the GNU Affero General Public License as published by
+//the Free Software Foundation, either version 3 of the License, or
+//(at your option) any later version.
+//
+//Talismane is distributed in the hope that it will be useful,
+//but WITHOUT ANY WARRANTY; without even the implied warranty of
+//MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//GNU Affero General Public License for more details.
+//
+//You should have received a copy of the GNU Affero General Public License
+//along with Talismane.  If not, see <http://www.gnu.org/licenses/>.
+//////////////////////////////////////////////////////////////////////////////
+package com.joliciel.talismane;
+
+import java.util.List;
+
+/**
+ * For getting notified whenever annotations are added.
+ * 
+ * @author Assaf Urieli
+ *
+ */
+public interface AnnotationObserver {
+	/**
+	 * An event fired whenever annotations are added to the observed subject.
+	 * 
+	 * @param subject
+	 *            the text to which annotations are about to be added
+	 * @param annotations
+	 *            the annotations getting added
+	 */
+	public <T> void beforeAddAnnotations(AnnotatedText subject, List<Annotation<T>> annotations);
```

Comment:
For now, all that can be done is add new annotations (not modify or delete existing ones): the state after add annotations is fully predictable, and the label "before" is simply there to make it clear that the existing list doesn't yet contain the annotations added.

The current use-case is when the annotated text is an abstraction (e.g. a sentence), we want to add the same annotations to a more concrete "container" (e.g. the raw text from which the sentence was extracted) - hence we need the subject passed, to allow the raw text to know the sentence's offset. To do this, it will of course have to treat the subject as a Sentence (rather than an AnnotatedText), which is a bit ugly.

However, an observer might be interested in the subject itself after the annotations have been added - in this case, the question is, should we also pass the annotations that have already been added to the subject's list.